File: app/src/server/jrpc_client.js
const axios = require('axios');

const PY_PORT = process.env.PY_PORT;
if (!PY_PORT) {
    throw new Error('PY_PORT not found');
}

const send=async (method, params) => {
    console.log("jrpc send",method,params)
    try {
        var url = `http://127.0.0.1:${PY_PORT}/jsonrpc`;
        const requestBody = {
            jsonrpc: '2.0',
            id: 1,
            method: method,
            params: params,
        };

        const response = await axios.post(url, requestBody);
        if (response.data.error) {
            throw new Error(`JSON-RPC Error: ${response.data.error.message}`);
        }

        return response.data.result;
    } catch (error) {
        console.error(`Error sending JSON-RPC request: ${error.message}`);
        return null;
    }
}

const ctxSend= async (method, ctx, args) => {
    const cid=global.cid(ctx);
    params={cid:cid,args:args};
    return await send(method,params);
}



const client = { send, ctxSend }



module.exports = client;

File: app/server/123/jrpc/jrpc_client.py
import requests
import json

import os
from dotenv import load_dotenv


load_dotenv(verbose=True)
JS_PORT=os.getenv("JS_PORT", "11082")


def send_jrpc_request(method, params):
    url=f"http://127.0.0.1:{JS_PORT}/jsonrpc"
    headers = {"content-type": "application/json"}

    payload = {
        "method": method,
        "params": params,
        "jsonrpc": "2.0",
        "id": 1,
    }

    response = requests.post(url, data=json.dumps(payload), headers=headers)

    if response.status_code == 200:
        response_data = response.json()

        if "result" in response_data:
            return response_data["result"]
        elif "error" in response_data:
            raise Exception("Error:", response_data["error"])
    else:
        raise Exception("Request failed with status code:", response.status_code)

def cid_send(method, cid, args):
    return send_jrpc_request(method,{"cid":cid,"args":args})
